go-controller/pkg/ovn/master.go

```diff
@@ -1259,19 +1246,42 @@ func (oc *Controller) syncNodesRetriable(nodes []interface{}) error {
 		if err := oc.deleteNode(nodeSwitch.Name, subnets); err != nil {
 			return fmt.Errorf("failed to delete node:%s, err:%v", nodeSwitch.Name, err)
 		}
-		//remove the node from the chassis map so we don't delete it twice
-		staleChassis.Delete(nodeSwitch.Name)
 	}
 
+	// cleanup stale chassis with no corresponding nodes
+	chassisList, err := libovsdbops.ListChassis(oc.sbClient)
+	if err != nil {
+		return fmt.Errorf("failed to get chassis list: %v", err)
+	}
+
+	chassisMap := map[string]*sbdb.Chassis{}
 	chassisDeleteList := []*sbdb.Chassis{}
 	for _, chassis := range chassisList {
-		if staleChassis.Has(chassis.Hostname) {
-			chassisDeleteList = append(chassisDeleteList, chassis)
+		chassisMap[chassis.Name] = chassis
+		// skip chassis that have a corresponding node
+		if foundNodes.Has(chassis.Hostname) {
+			continue
+		}
+		chassisDeleteList = append(chassisDeleteList, chassis)
+	}
+
+	// cleanup stale chassis private with no corresponding chassis
+	chassisPrivateList, err := libovsdbops.ListChassisPrivate(oc.sbClient)
+	if err != nil {
+		return fmt.Errorf("failed to get chassis private list: %v", err)
+	}
+
+	for _, chassis := range chassisPrivateList {
+		// skip chassis private that have a corresponding chassis
+		if _, found := chassisMap[chassis.Name]; found {
+			continue
 		}
+		chassisDeleteList = append(chassisDeleteList, &sbdb.Chassis{Name: chassis.Name})
```

Comment:
@jcaamano shouldnt' this be &sbdb.ChassisPrivate?

Reply:
Maybe instead of chassisMap containing all chassis, it should only contain chassis that we want to keep?

Reply:
Aha, ok. Thanks for the explanation. Might deserve a comment in master.go here?